widgets/hits.js

```diff
@@ -6,33 +6,40 @@ var bindProps = require('../decorators/bindProps');
 var Template = require('../components/Template');
 
 var defaultTemplates = {
-  empty: 'No matching objects, try another search',
-  hit: 'Object #{{objectID}}, this is the default `hits` template, you should provide one'
+  empty: 'No results',
+  hit: function(data) {
+    return JSON.stringify(data, null, 2);
+  }
 };
 
 /**
  * Display the list of results (hits) from the current search
  * @param  {String|DOMElement} options.container CSS Selector or DOMElement to insert the widget
  * @param  {Object} [options.templates] Templates to use for the widget
- * @param  {String|Function} [options.templates.empty=''] Template to use when there are no result
- * @param  {String|Function} [options.templates.hit=''] Template to use for each result
+ * @param  {String|Function} [options.templates.empty=''] Template to use when there are no results.
+ * Gets passed the `result` from the API call.
+ * @param  {String|Function} [options.templates.hit=''] Template to use for each result.
+ * Gets passed the `hit` of the result.
  * @param  {Object} [options.transformData] Method to change the object passed to the templates
  * @param  {Function} [options.transformData.empty=''] Method used to change the object passed to the empty template
  * @param  {Function} [options.transformData.hit=''] Method used to change the object passed to the hit template
- * @param  {boolean} [hideWhenNoResults=true] Hide the container when no results match
  * @param  {Number} [hitsPerPage=20] The number of hits to display per page
  * @return {Object}
  */
 function hits({
-    container = null,
+    container,
     templates = defaultTemplates,
     transformData,
-    hideWhenNoResults = false,
     hitsPerPage = 20
   }) {
   var Hits = require('../components/Hits');
 
   var containerNode = utils.getContainerNode(container);
+  var usage = 'Usage: hits({container, [templates.{empty,hit}, transformData.{empty,hit}, hitsPerPage])';
```

Comment:
I like the idea, but we'll need an automatic generator for that because the usage line really starts to get too big.